### DISCUSSION TITLE
Make self-hosted storage public - "New row violates RLS" error

### DISCUSSION THREAD

**Discussion by globemediaofficial** [supabase/supabase#9574] [Questions] [answered]
I understand that storage is still not included in the studio when self-hosting, even though a member of the team said they were working on it almost a year ago, so how can it be configured? I need the Row Level Security to be disabled because it needs to be public. Right now, even if a user is authenticated, trying to create a bucket in the app gives an error `New row violates RLS`.

**Comment by globemediaofficial** [comment, marked as answer]
1. Go to the `SQL Editor` tab on the studio dashboard
 2. `ALTER TABLE storage.buckets DISABLE ROW LEVEL SECURITY;`

Alternatively, you can create a policy. For some reason, row-level security is automatically enabled for the table but there is no default policy. Basically the policy doesn't exist so it will always be violated.

**Reply by Nisthar** [reply]
@globemediaofficial did you get realtime working with self host?

**Reply by globemediaofficial** [reply]
I haven’t tried using real-time yet but I have the database working.
